src/jit/compiler.h

```diff
@@ -1607,29 +1651,33 @@ struct fgArgTabEntry
             // Note that hfaSlots is the number of registers we will use. For ARM, that is twice
             // the number of "double registers".
             unsigned numHfaRegs = hfaSlots;
-            if (isPassedInRegisters())
-            {
 #ifdef _TARGET_ARM_
-                if (type == TYP_DOUBLE)
-                {
-                    // Must be an even number of registers.
-                    assert((numRegs & 1) == 0);
-                    numHfaRegs = hfaSlots / 2;
-                }
+            if (type == TYP_DOUBLE)
+            {
+                // Must be an even number of registers.
```

Comment:
Why does this this need to be even? Spec says "at most four uniquely addressable members"

Reply:
The must be homogeneous, and this is ARM32, where each double is passed in two registers.

Reply:
Ok, I missed the ARM32 part of this 👍